src/components/SiteHeader/HeaderLink.tsx

```diff
@@ -1,36 +1,67 @@
 import classNames from 'classnames'
-import { Link } from 'gatsby'
+import Link from 'gatsby-link'
 import _ from 'lodash'
 import React, { ReactNode } from 'react'
+import { Text } from 'rebass/styled-components'
+import styled from 'styled-components'
+
+import { HeaderTheme } from 'components/SiteHeader/SiteHeader'
+import useHover from 'utils/use-hover'
+
+const StyledHeaderLink = styled(Text)`
+  transition: color 0.3s;
+`
+
+const getIsActivePath = _.memoize(
+  ({ href, pathname }): boolean =>
+    (href === '/' && pathname === '/') ||
+    (href !== '/' && _.startsWith(pathname, href)),
+)
+
+const getThemeForPathname = (pathname: string): HeaderTheme =>
+  _.startsWith(pathname, '/photography') ? HeaderTheme.LIGHT : HeaderTheme.DARK
 
 interface Props {
-  children?: ReactNode
+  children: ReactNode
   className?: string
   href: string
-  isActive?: boolean
-  pathname?: string
+  pathname: string
 }
 
-const getIsActive = ({ href, pathname }): boolean =>
-  (href === '/' && pathname === '/') ||
-  (href !== '/' && _.startsWith(pathname, href))
+const HeaderLink = (props: Props) => {
+  const [isHovered, setIsHovered] = useHover()
+  const { children, className, href, pathname } = props
+  if (!pathname) {
+    return null
+  }
+
+  const isActive = getIsActivePath({ href, pathname })
+  const headerTheme = getThemeForPathname(pathname)
 
-const HeaderLink = ({ children, className, href, pathname }: Props) => {
-  const isActive = getIsActive({ href, pathname })
   return (
-    <Link
+    <StyledHeaderLink
+      {...setIsHovered}
+      as={Link}
+      to={href}
       className={classNames(
-        'shadow-none flex justify-center items-center lato f4 ph3 mv2 pv2 flex-auto light-silver fw3',
-        className,
+        'f4 fw3 mv2 ph3 pv2 shadow-none',
         {
-          'light-silver fw3': !isActive,
-          'white-90 fw4': isActive,
+          'black-80': isHovered && headerTheme === HeaderTheme.LIGHT,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 20 spaces but found 10. (@typescript-eslint/indent)](https://app.codacy.com/app/jdstuart/jesses.io/pullRequest?prid=4576429)